staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -113,6 +113,20 @@ type Config struct {
 	// Version forces a specific version to be used (if registered)
 	// Do we need this?
 	// Version string
+
+	// BearerTokenSource for managing the token
+	BearerTokenSource BearerTokenSource
+}
+
+// BearerTokenSource is used to generate
+type BearerTokenSource interface {
+
+	// Token returns a bearer token to add to all requests sent by the REST client.
+	Token() (string, error)
+
+	// Refresh is triggered when the client experiences a 401, indicating the current
+	// bearer token is invalid and should be renewed.
+	Refresh() (ok bool, err error)
```

Comment:
is the caller or the token source responsible for managing the rate or number of retries on 401 responses?